packages/testing-library/react-lynx-testing-library/src/__tests__/events.test.jsx

```diff
@@ -0,0 +1,165 @@
+import '@testing-library/jest-dom';
+import { test } from 'vitest';
+import { fireEvent, render } from '..';
+import { createRef } from '@lynx-js/react';
+import { expect, vi } from 'vitest';
+
+const eventTypes = [
+  {
+    type: 'LynxBindCatchEvent',
+    events: [
+      'tap',
+      'longtap',
+    ],
+    init: {
+      key: 'value',
+    },
+  },
+  {
+    type: 'LynxEvent',
+    events: [
+      'bgload',
+      'bgerror',
+      'touchstart',
+      'touchmove',
+      'touchcancel',
+      'touchend',
+      'longpress',
+      'transitionstart',
+      'transitioncancel',
+      'transitionend',
+      'animationstart',
+      'animationiteration',
+      'animationcancel',
+      'animationend',
+      'mousedown',
+      'mouseup',
+      'mousemove',
+      'mouseclick',
+      'mousedblclick',
+      'mouselongpress',
+      'wheel',
+      'keydown',
+      'keyup',
+      'focus',
+      'blur',
+      'layoutchange',
+    ],
+  },
+];
+
+eventTypes.forEach(({ type, events, elementType, init }, eventTypeIdx) => {
+  describe(`${type} Events`, () => {
+    events.forEach((eventName, eventIdx) => {
+      const eventProp = `bind${eventName}`;
+
+      it(`triggers ${eventProp}`, async () => {
+        const ref = createRef();
+        const spy = vi.fn();
+
+        const Comp = () => {
+          return (
+            <view
+              ref={ref}
+              {...{
+                [eventProp]: spy,
+              }}
+            />
+          );
+        };
+
+        render(<Comp />);
+
+        if (eventTypeIdx === 0 && eventIdx === 0) {
+          expect(ref).toMatchInlineSnapshot(`
+            {
+              "current": NodesRef {
+                "_nodeSelectToken": {
+                  "identifier": "1",
+                  "type": 2,
+                },
+                "_selectorQuery": {},
+              },
+            }
+          `);
+          expect(ref.current.constructor.name).toMatchInlineSnapshot(
+            `"NodesRef"`,
+          );
+          const element = __GetElementByUniqueId(
+            Number(ref.current._nodeSelectToken.identifier),
+          );
+          expect(element).toMatchInlineSnapshot(`<view />`);
```

Comment:
Seems `ref` and `event` are completely omitted in snapshot?

Can we somehow make it in snapshot? For easier testing...

Reply:
One can simply use `element.props` to do asserts on `ref` and `event`, let me write a document for this:

```js
expect(element).toMatchInlineSnapshot(`<view />`);
expect(element.props).toMatchInlineSnapshot(`
  {
    "event": {
      "bindEvent:tap": [Function],
    },
    "has-react-ref": true,
  }
`)
```

Reply:
@hzy Added examples in our docs: https://github.com/lynx-family/lynx-website/pull/117